docs/docs/external-volumes

```diff
@@ -0,0 +1,131 @@
+
+- The `external.driver["dvdi/driver"]` option specifies which Docker volume driver to use for storage. If you are running Marathon on DCOS, this value should always be `rexray`. [Learn more about REX-Ray](https://rexray.readthedocs.org/en/v0.3.2/user-guide/schedulers/).
+
+- You can specify additional options with `container.volumes[x].external.options[optionName]`. The dvdi provider for Mesos containers uses `dvdcli`, which offers the options [documented here[(https://github.com/emccode/dvdcli#extra-options). The availability of any given option depends on your volume driver, however.
+
+- `name` is the name by which your volume driver looks up your volume. When your task is staged on an agent, the volume driver queries the storage service for a volume with this name. If one does not exist, it's created. Otherwise, the existing volume is re-used. **Note:** Implicit volume creation only works when using volumes with a Mesos container.
+
+- Create multiple volumes by adding additional items in the `container.volumes` array.
+
+- Volume parameters cannot be changed after you create the application.
+
+- **Note:** Marathon will not launch apps with external volumes if  `upgradeStrategy.minimusHealthCapacity` is less than 0.5, or if `upgradeStrategy.maximumOverCapacity` does not equal 0.
+
+### Using a Docker Container
+
+Below is a sample app definition that uses a Docker container and specifies an external volume:
+
+```json
+{
+  "id": "/test-docker",
+  "instances": 1,
+  "cpus": 0.1,
+  "mem": 32,
+  "cmd": "/usr/bin/tail -f /dev/null",
+  "container": {
+    "type": "DOCKER",
+    "docker": {
+      "image": "alpine:3.1",
+      "network": "HOST",
+      "forcePullImage": true
+    },
+    "volumes": [
+      {
+        "containerPath": "/data/test-rexray-volume",
+        "external": {
+          "size": 100,
+          "name": "my-test-vol",
+          "provider": "dvdi",
+          "options": { "dvdi/driver": "rexray" }
+        },
+        "mode": "RW"
+      }
+    ]
+  },
+  "upgradeStrategy": {
+    "minimumHealthCapacity": 0,
+    "maximumOverCapacity": 0
+  }
+}
+```
+
+For more information, refer to the [REX-Ray documentation](https://rexray.readthedocs.org/en/v0.3.2/user-guide/schedulers/#docker-containerizer-with-marathon).
+
+## Potential Pitfalls
+
+- You can only assign one task per volume. Your storage provider may other particular limitations.
+
+- The volumes you create are not automatically cleaned up. If you you delete your cluster, go to your storage provider and delete the volumes you no longer need. If you using EBS, find them by searching by the `container.volume.external.name` that you set in your Marathon app definition. This name corresponds to an EBS volume `Name` tag.
+
+- Volumes are not isolated from other clusters. Choose a unique volume name to avoid conflicts.
```

Comment:
we also noticed that, when using CCM, clusters can be spun up in different AWS availability zones. for example I could create a cluster and use some external vols -- and CCM picked AZ1 for this cluster. I then destroy the cluster. I then launch a new cluster. CCM picks AZ2. I try to use the external vol from my first cluster but I can't because my new cluster is in a different AZ.

This may be a cloud-provider-specific "gotcha" but it was something that bit me during testing. I'm sure I won't be the only one.